modules/home-age.nix

```diff
@@ -0,0 +1,154 @@
+    '')
+    cfg.identityPaths;
+
+  installSecrets = builtins.concatStringsSep "\n" ([ "echo '[agenix] decrypting user secrets...'" ] ++ (map installSecret (builtins.attrValues cfg.secrets)));
+
+  secretType = types.submodule ({ config, ... }: {
+    options = {
+      name = mkOption {
+        type = types.str;
+        default = config._module.args.name;
+        description = ''
+          Name of the file used in ''${cfg.secretsDir}
+        '';
+      };
+      file = mkOption {
+        type = types.path;
+        description = ''
+          Age file the secret is loaded from.
+        '';
+      };
+      path = mkOption {
+        type = types.str;
+        default = "${cfg.secretsDir}/${config.name}";
+        description = ''
+          Path where the decrypted secret is installed.
+        '';
+      };
+      mode = mkOption {
+        type = types.str;
+        default = "0400";
+        description = ''
+          Permissions mode of the decrypted secret in a format understood by chmod.
+        '';
+      };
+      # Don't know how to make it find config.home.username
+      #  owner = mkOption {
+      #    type = types.str;
+      #    readOnly = true;
+      #    default = "${config.home.username}";
+      #    description = ''
+      #      User of the decrypted secret.
+      #    '';
+      #  };
+      group = mkOption {
+        type = types.str;
+        readOnly = true;
+        default = "$(id -gn)";
+        description = ''
+          Group of the decrypted secret.
+        '';
+      };
+      symlink = mkEnableOption "symlinking secrets to their destination" // { default = true; };
+    };
+  });
+in
+{
+  options.age = {
+    ageBin = mkOption {
+      type = types.str;
+      default = "${rage}/bin/rage";
+      description = ''
+        The age executable to use.
+      '';
+    };
+    secrets = mkOption {
+      type = types.attrsOf secretType;
+      default = { };
+      description = ''
+        Attrset of secrets.
+      '';
+    };
+    secretsDir = mkOption {
+      type = types.str;
+      readOnly = true;
+      default = "/run/user/$(id -u)/agenix"; # Need to figure something out with the runtime dir
+      description = ''
+        Folder where secrets are symlinked to
+      '';
+    };
+    secretsMountPoint = mkOption {
+      type = types.addCheck types.str
+        (s:
+          (builtins.match "[ \t\n]*" s) == null # non-empty
+            && (builtins.match ".+/" s) == null) # without trailing slash
+      // { description = "${types.str.description} (with check: non-empty without trailing slash)"; };
+      readOnly = true;
+      default = "/run/user/$(id -u)/agenix.d"; # here too
+      defaultText = "/run/user/$(id -u)/agenix.d";
+      description = ''
+        Where secrets are created before they are symlinked to ''${cfg.secretsDir}
+      '';
+    };
+    identityPaths = mkOption {
+      type = types.listOf types.path;
+      default = [ "${config.home.homeDirectory}/.ssh" ];
```

Comment:
`identityPaths` needs to be the full path of the decryption key, I believe.

Reply:
Fixed.